sql/sql_statistics.cc

```diff
@@ -2608,18 +2641,18 @@ bool Column_statistics_collected::add()
 
 
 /* 
-  Create an empty Histogram object from histogram_type.
+  Create an empty Histogram_binary object from histogram_type.
 
   Note: it is not yet clear whether collection-time histogram should be the same 
   as lookup-time histogram. At the moment, they are.
 */
 
-Histogram* get_histogram_by_type(MEM_ROOT *mem_root, Histogram_type hist_type) {
+Histogram_binary * get_histogram_by_type(MEM_ROOT *mem_root, Histogram_type hist_type) {
   switch (hist_type) {
   case SINGLE_PREC_HB:
   case DOUBLE_PREC_HB:
   case JSON:
-    return new Histogram();
+    return new Histogram_binary();
   default:
```

Comment:
Why is Histogram_binary created when type=JSON? 
.. also I see that the function is not used anywhere...